### PR TITLE
EA-3680 fix error messages

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -65,6 +64,14 @@ public ModelAndView europeanaExceptionHandler(HttpServletRequest request, HttpSe
...
     }
 
+    @ExceptionHandler(value = {HttpException.class})
+    public ModelAndView httpExceptionHandler(HttpServletRequest request, HttpServletResponse response, HttpException ee) {
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        return generateErrorResponse(request, response, "Unauthorized",
+                I18nErrorMessageKeys.getMessageForKey(ee.getI18nKey()), StringUtils.substringAfter(ee.getI18nKey(), "."));
+    }
+
+
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java:74]: It looks like we are sending unauthorized whenever there is a problem sending a request over http. However, in the past we agreed that whenever we can't communicate with the API key service, we'll silently allow the request and just log that we were not able to verify the key. 
Exception is when there is no key (or token) at all, in those cases we'll always block the request.

Comment (SrishtiSingh-eu) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java:74]: yes, all the HttpException regarding authentication is coming via [BaseAuthorizationService](https://github.com/europeana/api-commons/blob/master/commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java) class.
I tested the flow without the apikey url, it by-passes the validation of the API key.
 
The Apikey silent approval will work as well as after extracting Apikey when we validate the Apikey and get any exceptions, we are silently approving it without throwing any exceptions. ([here](https://github.com/europeana/api-commons/blob/master/commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java#L83))

Also, this is a custom HttpException class - https://github.com/europeana/api-commons/blob/master/commons-web/src/main/java/eu/europeana/api/commons/web/exception/HttpException.java#L10
right now only the ApplicationAuthenticationException exception extends the HttpException class
So only if we add another exception that extends HttpException we will have any issue. So far all the api's use the HttpException handling present in GlobalExceptionHandler. Which is similar to what we do here.

Comment (gsergiu) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java:74]: actually GlobalExceptionHandler should extend https://github.com/europeana/api-commons/blob/3b7a7dd01cb956b48c50c0be235755aa155843b9/commons-web/src/main/java/eu/europeana/api/commons/web/exception/EuropeanaGlobalExceptionHandler.java which implements the updated specs for the error responses and handles the serialization of all base exception classes. This is already used in the translation API https://github.com/europeana/translation-api/blob/develop/translation-web/src/main/java/eu/europeana/api/translation/web/exception/GlobalExceptionHandler.java

Comment (gsergiu) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java:74]: I think that the ClientRegistrationException is thrown when we get the invalid apikey response here: https://github.com/europeana/api-commons/blob/3b7a7dd01cb956b48c50c0be235755aa155843b9/commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java#L87, and when we communication failed we should get the other exception on which we silently approve the key: https://github.com/europeana/api-commons/blob/3b7a7dd01cb956b48c50c0be235755aa155843b9/commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java#L92C22-L92C33

I'm not sure if we have some issues whre raising those exeption in the APIKeyClient

Comment (SrishtiSingh-eu) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/GlobalExceptionHandler.java:74]: The problem is even after adding Bean for I18Service and message.en Properties file the values for the error messages are not picked up.

**Pull request merged**